### DISCUSSION TITLE
How to call certain functions only when log level is DEBUG in plugin?

### DISCUSSION THREAD

**Discussion by amitonlentra** [Kong/kong#13474] [Help] [answered]
We are deploying kong 3.1.1 through helm on kubernetes in our environments.

For performance reasons we want to restrict the call to certain modules and functions only when the log level is DEBUG.

How do get to know the log level in the kong plugin handler.lua file?

The logging is done through
`ngx.log(ngx.DEBUG, "encrypted data: ", resty_str.to_hex(data))`

I tried `os.getenv("KONG_LOG_LEVEL")` but that returns nil even when the log level was configured through an environment variable. I am locally running kong as a docker container but that shouldn’t have any impact I think.

Kindly share your expert inputs.

Thanks.

**Comment by ProBrian** [comment]
How did you set the env environment in your deployment? could you try testing with other deployment method like local build from source, to make sure it's a functional issue or something wrong in k8s deployment logic?

**Comment by bungle** [comment, marked as answer]
You can use:
```lua
if kong.configuration.log_level == "debug" then
  ...
end
```

The env vars with nginx are only available on `init` phase, but not with workers. You can also place `os.getenv` on module level of your `handler.lua` as handlers are loaded on init phase. But I think it is better to use above.

You obviously don't need to wrap any `ngx.log` or `kong.log.debug`  with that.

**Reply by amitonlentra** [reply]
Works as expected. Thanks.
